presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1100,6 +1107,25 @@ public ConnectorTableLayout getTableLayout(ConnectorSession session, ConnectorTa
                 ImmutableList.of());
     }
 
+    private static <T> TupleDomain<T> columnWiseUnion(Set<T> columns, Iterable<TupleDomain<T>> tupleDomains)
+    {
+        Map<T, Domain> domainsByColumn = new HashMap<>(columns.size());
+
+        for (TupleDomain<T> tupleDomain : tupleDomains) {
+            Optional<Map<T, Domain>> domains = tupleDomain.getDomains();
+            if (domains.isPresent()) {
```

Comment:
you should assert that the domains.isPresent() (not present means a None TupleDomain), if that is ever not the case, then this will produce incorrect results.